### PR TITLE
Add source .proto file for update_metadata_pb2.py

### LINKED ISSUE #58: TypeError: __new__() got an unexpected keyword argument 'serialized_options'
Apparently this error can occur when [the host's protobuf runtime doesn't match the compiled protobuf file](https://github.com/protocolbuffers/protobuf/issues/4716):

```
[*] Using debugfs for image mounts
[*] Extracting '/home/nick/android11/android-prepare-vendor/out/blueline/rq1d.210205.004/blueline-rq1d.210205.004-factory-3666d140.zip'
[*] Unzipping 'image-blueline-rq1d.210205.004.zip'
[*] Extracting '/home/nick/android11/android-prepare-vendor/out/blueline/rq1d.210205.004/blueline-ota-rq1d.210205.004-23e968d6.zip'
Traceback (most recent call last):
  File "/home/nick/android11/android-prepare-vendor/scripts/extract_android_ota_payload/extract_android_ota_payload.py", line 13, in <module>
    import update_metadata_pb2
  File "/home/nick/android11/android-prepare-vendor/scripts/extract_android_ota_payload/update_metadata_pb2.py", line 23, in <module>
    serialized_pb=_b('\n\x15update_metadata.proto\x12\x16\x63hromeos_update_engine\"1\n\x06\x45xtent\x12\x13\n\x0bstart_block\x18\x01 \x01(\x04\x12\x12\n\nnum_blocks\x18\x02 \x01(\x04\"z\n\nSignatures\x12@\n\nsignatures\x18\x01 \x03(\x0b\x32,.chromeos_update_engine.Signatures.Signature\x1a*\n\tSignature\x12\x0f\n\x07version\x18\x01 \x01(\r\x12\x0c\n\x04\x64\x61ta\x18\x02 \x01(\x0c\"+\n\rPartitionInfo\x12\x0c\n\x04size\x18\x01 \x01(\x04\x12\x0c\n\x04hash\x18\x02 \x01(\x0c\"w\n\tImageInfo\x12\r\n\x05\x62oard\x18\x01 \x01(\t\x12\x0b\n\x03key\x18\x02 \x01(\t\x12\x0f\n\x07\x63hannel\x18\x03 \x01(\t\x12\x0f\n\x07version\x18\x04 \x01(\t\x12\x15\n\rbuild_channel\x18\x05 \x01(\t\x12\x15\n\rbuild_version\x18\x06 \x01(\t\"\xee\x03\n\x10InstallOperation\x12;\n\x04type\x18\x01 \x02(\x0e\x32-.chromeos_update_engine.InstallOperation.Type\x12\x13\n\x0b\x64\x61ta_offset\x18\x02 \x01(\x04\x12\x13\n\x0b\x64\x61ta_length\x18\x03 \x01(\x04\x12\x33\n\x0bsrc_extents\x18\x04 \x03(\x0b\x32\x1e.chromeos_update_engine.Extent\x12\x12\n\nsrc_length\x18\x05 \x01(\x04\x12\x33\n\x0b\x64st_extents\x18\x06 \x03(\x0b\x32\x1e.chromeos_update_engine.Extent\x12\x12\n\ndst_length\x18\x07 \x01(\x04\x12\x18\n\x10\x64\x61ta_sha256_hash\x18\x08 \x01(\x0c\x12\x17\n\x0fsrc_sha256_hash\x18\t \x01(\x0c\"\xad\x01\n\x04Type\x12\x0b\n\x07REPLACE\x10\x00\x12\x0e\n\nREPLACE_BZ\x10\x01\x12\x0c\n\x04MOVE\x10\x02\x1a\x02\x08\x01\x12\x0e\n\x06\x42SDIFF\x10\x03\x1a\x02\x08\x01\x12\x0f\n\x0bSOURCE_COPY\x10\x04\x12\x11\n\rSOURCE_BSDIFF\x10\x05\x12\x0e\n\nREPLACE_XZ\x10\x08\x12\x08\n\x04ZERO\x10\x06\x12\x0b\n\x07\x44ISCARD\x10\x07\x12\x11\n\rBROTLI_BSDIFF\x10\n\x12\x0c\n\x08PUFFDIFF\x10\t\"\xd7\x05\n\x0fPartitionUpdate\x12\x16\n\x0epartition_name\x18\x01 \x02(\t\x12\x17\n\x0frun_postinstall\x18\x02 \x01(\x08\x12\x18\n\x10postinstall_path\x18\x03 \x01(\t\x12\x17\n\x0f\x66ilesystem_type\x18\x04 \x01(\t\x12M\n\x17new_partition_signature\x18\x05 \x03(\x0b\x32,.chromeos_update_engine.Signatures.Signature\x12\x41\n\x12old_partition_info\x18\x06 \x01(\x0b\x32%.chromeos_update_engine.PartitionInfo\x12\x41\n\x12new_partition_info\x18\x07 \x01(\x0b\x32%.chromeos_update_engine.PartitionInfo\x12<\n\noperations\x18\x08 \x03(\x0b\x32(.chromeos_update_engine.InstallOperation\x12\x1c\n\x14postinstall_optional\x18\t \x01(\x08\x12=\n\x15hash_tree_data_extent\x18\n \x01(\x0b\x32\x1e.chromeos_update_engine.Extent\x12\x38\n\x10hash_tree_extent\x18\x0b \x01(\x0b\x32\x1e.chromeos_update_engine.Extent\x12\x1b\n\x13hash_tree_algorithm\x18\x0c \x01(\t\x12\x16\n\x0ehash_tree_salt\x18\r \x01(\x0c\x12\x37\n\x0f\x66\x65\x63_data_extent\x18\x0e \x01(\x0b\x32\x1e.chromeos_update_engine.Extent\x12\x32\n\nfec_extent\x18\x0f \x01(\x0b\x32\x1e.chromeos_update_engine.Extent\x12\x14\n\tfec_roots\x18\x10 \x01(\r:\x01\x32\"L\n\x15\x44ynamicPartitionGroup\x12\x0c\n\x04name\x18\x01 \x02(\t\x12\x0c\n\x04size\x18\x02 \x01(\x04\x12\x17\n\x0fpartition_names\x18\x03 \x03(\t\"s\n\x18\x44ynamicPartitionMetadata\x12=\n\x06groups\x18\x01 \x03(\x0b\x32-.chromeos_update_engine.DynamicPartitionGroup\x12\x18\n\x10snapshot_enabled\x18\x02 \x01(\x08\"\xb1\x06\n\x14\x44\x65ltaArchiveManifest\x12\x44\n\x12install_operations\x18\x01 \x03(\x0b\x32(.chromeos_update_engine.InstallOperation\x12K\n\x19kernel_install_operations\x18\x02 \x03(\x0b\x32(.chromeos_update_engine.InstallOperation\x12\x18\n\nblock_size\x18\x03 \x01(\r:\x04\x34\x30\x39\x36\x12\x19\n\x11signatures_offset\x18\x04 \x01(\x04\x12\x17\n\x0fsignatures_size\x18\x05 \x01(\x04\x12>\n\x0fold_kernel_info\x18\x06 \x01(\x0b\x32%.chromeos_update_engine.PartitionInfo\x12>\n\x0fnew_kernel_info\x18\x07 \x01(\x0b\x32%.chromeos_update_engine.PartitionInfo\x12>\n\x0fold_rootfs_info\x18\x08 \x01(\x0b\x32%.chromeos_update_engine.PartitionInfo\x12>\n\x0fnew_rootfs_info\x18\t \x01(\x0b\x32%.chromeos_update_engine.PartitionInfo\x12\x39\n\x0eold_image_info\x18\n \x01(\x0b\x32!.chromeos_update_engine.ImageInfo\x12\x39\n\x0enew_image_info\x18\x0b \x01(\x0b\x32!.chromeos_update_engine.ImageInfo\x12\x18\n\rminor_version\x18\x0c \x01(\r:\x01\x30\x12;\n\npartitions\x18\r \x03(\x0b\x32\'.chromeos_update_engine.PartitionUpdate\x12\x15\n\rmax_timestamp\x18\x0e \x01(\x03\x12T\n\x1a\x64ynamic_partition_metadata\x18\x0f \x01(\x0b\x32\x30.chromeos_update_engine.DynamicPartitionMetadataB\x02H\x03')
TypeError: __new__() got an unexpected keyword argument 'serialized_options'
[-] OTA data extract failed
```

I don't see a `update_metadata.proto` source file checked into the APV repository; maybe it is not a good idea to be committing just the generated file, with no easy way to regenerate it to match the installed protobuf libraries.

### PR DESCRIPTION
**Pull request by nickrbogdanov**
*_pb2.py files are autogenerated by the protobuf compiler.  Sometimes
the format changes in a way that is incompatible with the
python-protobuf runtime, so it is unwise to maintain *just* the
compiled version in our tree.  Add the corresponding source file and
a Makefile (borrowed from carriersettings-extractor) to let the user
easily regenerate it from source.

I found update_metadata.proto in this repo:
https://android.googlesource.com/platform/system/update_engine

By comparing the compiled output I determined that our current version
of update_metadata_pb2.py contains the deprecation flags introduced in
upstream commit 0f59a9a41177, and it does not contain the
unpadded_signature_size field added in the subsequent commit
(7bbe015a1bd1c).  So I synced to 0f59a9a41177.

Fixes AOSPAlliance/android-prepare-vendor#58.

### CODE REVIEW

**Comment by dan-v** [comment]
@nickrbogdanov - thanks for this contribution! What version of protobuf did you have installed when you were bumping into #58? It would be nice if we could at least specify what version(s) of python-protobuf are compatible with the currently generated update_metadata_pb2.py.

Not necessary for this PR, but it also might be nice to add a Dockerfile with a pinned version of protobuf that could be used to regenerate these files, as the generated output appears to vary based on the version of protoc.

**Comment by nickrbogdanov** [comment]
I have version 3.0.0-9.1ubuntu1 and it is not compatible with the current version of `update_metadata_pb2.py`.

I think that if we're willing to require users to install an external dependency (protobuf compiler), any modern version of the compiler + runtime should be fine as long as the generated code is compatible with the installed python library.  Maybe part of the problem here is that `python-protobuf` is a dependency but it isn't mentioned in the README?  We could just ask users to install both packages and then regenerate from the .proto file every time.

**Comment by dan-v** [comment]
Thanks for the additional details. @nickrbogdanov. 

> Maybe part of the problem here is that `python-protobuf` is a dependency but it isn't mentioned in the README

Agreed. I think it makes sense to call out these dependencies and probably makes sense to regenerate from proto file as part of the normal script execution. I wouldn't want to force generation of these python files without it being nicely integrated, as that changes the workflow to something like "first run make ...".

Anyways, for now, the changes you have made at least allow people to regenerate which is a first good step. I'm going to approve and wait for one other person to stamp this before merge.

**Review by dan-v — APPROVED** [review]

**Comment by dan-v** [comment]
@chirayudesai this could use a look when you get a chance

**Comment by chirayudesai** [comment]
This looks fine. Re-generating via the makefile gets rid of the python2 compat present in the files currently but that's fine given that python2 has been deprecated anyway.

Thanks!

**Pull request merged**
